### DISCUSSION TITLE
[Idea] Manual mode on the ps_bestsellers module

### DISCUSSION THREAD

**Discussion by Lionel-dev** [PrestaShop/PrestaShop#31638] [Ideas]
### The problem

Many of my customers are a bit surprised by the products that come out on the front. 

Often, they are not happy with the products put forward, probably, because they are not expensive or not representative enough of their shop (e.g.: some customers had batteries as products on the home, while they sell models...)

### The solution

Propose a manual mode to enter the ids of the products you want to highlight on this block. 

Like this (I already implemented this feature on my side, but I would like to propose it in PR) :

![image](https://user-images.githubusercontent.com/48990561/222740372-cc598bb5-5ef1-4c8a-ba18-ea4be3af9434.png)

### Do you plan to work on this subject?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

**Comment by kpodemski** [comment]
It's a common ask from the merchants. I agree. I usually create a clone of the `ps_featuredproducts` and allow them to create a special category to put Bestsellers into and provide its ID in the configuration. Would providing only IDs (without any actual search) be a good UX for less technical merchants?

I like the idea tho! I'll ping some other project members. What do you think @Hlavtox @MatShir @lmeyer1 ?

**Reply by lmeyer1** [reply]
Yes, I think it is common, that the statistical bestsellers are not those that the merchant would like to see. 

Is it appropriate to add a manual product choice into the automatic bestsellers module ? I don't think so. Using `ps_featuredproduct` instead seems cleaner to me.

Is a list of product IDs a good enough UX? It is good enough for my internally used modules, but it seems to me that official Prestashop modules usually use some more evolved way to choose products. On the other hand, if there are many products (several tens of thousands), then entering product IDs is actually faster than a product search (a product search like on the BO order page is painful, and a product search like the one pour cart rules is unuseable).

**Comment by Hlavtox** [comment]
I like the idea, but then it's more of a `ps_featuredproducts` block, where you can select products from selected category. :-)))
We should not mess with the behavior of bestsellers.

For featured products, I am all in. Let's add a selectbox like this and we are golden.
`Which products you want to display`
`Products from selected categories` or `Manually selected products`

For products, we could use a TypeAhead input like in product page accessories.

**Reply by Lionel-dev** [reply]
I think your proposal is really perfect!

**Reply by web-cooking-factory** [reply]
I think it's the right solution.
Even more, be able to create multiple blocks like that.

**Comment by Lionel-dev** [comment]
Sorry for my delay in answering, I had not seen the notification ^^

Indeed it's quite basic, but easily modifiable with some classes provided in prestashop, we could easily do something like this : 

![image](https://user-images.githubusercontent.com/48990561/224051517-334a5be7-6cb6-45a2-8237-f511f134575a.png)

By adding a multiple select, of course it requires several adaptations in backend, but nothing insurmountable if the idea likes.

**Reply by Hlavtox** [reply]
We will need to use something that fetches it from the database, imagine we load 20k products into a selectbox, it would explode :D

**Reply by Lionel-dev** [reply]
You're right!

**Reply by lmeyer1** [reply]
> It would explode

Like on the cart rules page: such a select list takes about 299 seconds to load, so there remains 1 second for the merchant to choose before the timeout 🤣. This is what I meant by `unuseable` above.
